enhancements/monitoring/replace-prometheus-adapter-with-metrics-server.md

```diff
@@ -0,0 +1,252 @@
+    // Defines the nodes on which the pods are scheduled.
+    NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+    // Defines tolerations for the pods.
+    Tolerations []v1.Toleration `json:"tolerations,omitempty"`
+    // Defines resource requests and limits for the Prometheus container.
+    Resources *v1.ResourceRequirements `json:"resources,omitempty"`
+  }
+  ```
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Configuration/Enablement
+
+Metrics Server is not enabled by default. It must be explicitly enabled by the user. It is enabled by setting 
+`enabled: true` in the ConfigMap `openshift-monitoring/cluster-monitoring-config` under `metricsServer` section
+
+#### Resource Impact
+
+Once metrics-server is enabled, by default 2 replicas of `metrics-server` instances are being deployed in the openshift-monitoring namespace.
+
+Once metrics-server is fully functional (after 5 scrape cycles done by metrics-server) we will remove `prometheus-adapter` resources fully.
+
+### Risks and Mitigations
+
+When considering equal scrape intervals, both metrics-server and prometheus-adapter have a comparable impact. This similarity arises from the fact that both applications perform a kubelet scrape once per instance within the designated scrape interval. We compared cpu and memory usage of metric-server vs prometheus-adapter on cluster-bot's cluster in [MON-3210](https://issues.redhat.com/browse/MON-3210) which didnt see significant difference when prometheus-adapter was replaced with metrics-server.
+
+We need to do performance testing on a larger OpenShift cluster (more than 100 nodes) and more than 70 pods on per node to measure the performance of Metrics Server. This is based on scalability table mentioned in [metrics-server](https://github.com/kubernetes-sigs/metrics-server#scaling). [MON-3328](https://issues.redhat.com/browse/MON-3328) is created for doing this.
+
+### Drawbacks
+
+N/A
+
+
+## Design Details
+
+### Open Questions [optional]
+
+### Test Plan
+
+The proposed solution will be tested with two suites:
+
+* Unit tests - running independently of each other and without side effects
+* End to end tests - in the same repository as the cluster-monitoring-operator. These tests will be executed against an actual Kubernetes cluster and are intended to verify end-to-end functionality of the entire system.
+
+E2E Test will test following:
+  * Enables metrics-server
+  * Verifies that the resource metrics API is working
+  * Disables metrics-server
+  * Verifies that the resource metrics API is working
+  * During all the test, it will ensure that there's no disruption of the resource metrics API.
+
+### Graduation Criteria
+
+The main rationale behind adopting a phased approach is to prevent customers from encountering problems with the metrics-server without a contingency plan in place. This approach ensures a mechanism for reverting to the previous/legacy deployment if needed.
+
+* 4.15 TechPreview
```

Comment:
I think that we can even negotiate GA for 4.15 (as opt-in) if we manage to merge soon enough in the cycle.